server/http.go

```diff
@@ -37,180 +42,86 @@ func (p *Plugin) ServeHTTP(c *plugin.Context, w http.ResponseWriter, r *http.Req
-//			return
-//		}
-//		meetingID = rm.ID
-//	}
-//
-//	//url := config.ZoomURL
-//	url := ""
-//	zoomURL := strings.TrimSpace(url)
-//	if len(zoomURL) == 0 {
-//		zoomURL = "https://zoom.us"
-//	}
-//
-//	meetingURL := fmt.Sprintf("%s/j/%v", zoomURL, meetingID)
-//
-//	post := &model.Post{
-//		UserId:    p.botUserID,
-//		ChannelId: req.ChannelID,
-//		Message:   fmt.Sprintf("Meeting started at %s.", meetingURL),
-//		Type:      "custom_zoom",
-//		Props: map[string]interface{}{
-//			"meeting_id":       meetingID,
-//			"meeting_link":     meetingURL,
-//			"meeting_status":   webex.WebhookStatusStarted,
-//			"meeting_personal": personal,
-//			"meeting_topic":    req.Topic,
-//		},
-//	}
-//
-//	createdPost, appErr := p.API.CreatePost(post)
-//	if appErr != nil {
-//		http.Error(w, appErr.Error(), appErr.StatusCode)
-//		return
-//	}
-//
-//	if appErr = p.API.KVSet(fmt.Sprintf("%v%v", postMeetingKey, meetingID), []byte(createdPost.Id)); appErr != nil {
-//		http.Error(w, appErr.Error(), appErr.StatusCode)
-//		return
-//	}
-//
-//	if _, err := w.Write([]byte(fmt.Sprintf("%v", meetingID))); err != nil {
-//		p.API.LogWarn("failed to write response", "error", err.Error())
-//	}
-//}
+type startMeetingRequest struct {
+	ChannelID string `json:"channel_id"`
+	MeetingID int    `json:"meeting_id"`
+}
+
+func (p *Plugin) handleStartMeeting(w http.ResponseWriter, r *http.Request) (int, error) {
+	//config := p.getConfiguration()
+
+	userId := r.Header.Get("Mattermost-User-Id")
+	if userId == "" {
+		return http.StatusUnauthorized, errors.New("not authorized")
+	}
+
+	var req startMeetingRequest
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		return http.StatusBadRequest, fmt.Errorf("err: %v", err)
+	}
+
+	if _, appErr := p.API.GetChannelMember(req.ChannelID, userId); appErr != nil {
+		return http.StatusForbidden, errors.New("forbidden")
+	}
+
+	roomId, err := p.getRoomOrDefault(userId)
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	siteURL := p.getConfiguration().SiteHost
+	if siteURL == "" {
+		return http.StatusInternalServerError, errors.New("Unable to setup a meeting; the Webex plugin has not been configured yet.\nPlease ask your system administrator to set the `Webex Site Hostname` in `System Console -> PLUGINS -> Webex`.")
+	}
+
+	webexJoinURL := (&url.URL{
+		Scheme: "https",
+		Host:   siteURL,
+		Path:   "/join/" + roomId,
+	}).String()
+
+	webexStartURL := (&url.URL{
+		Scheme: "https",
+		Host:   siteURL,
+		Path:   "/start/" + roomId,
+	}).String()
+
+	joinPost := &model.Post{
+		UserId:    p.botUserID,
+		ChannelId: req.ChannelID,
+		Message:   fmt.Sprintf("Meeting started at %s.", webexJoinURL),
+		Type:      "custom_webex",
+		Props: map[string]interface{}{
+			"meeting_link":     webexJoinURL,
+			"meeting_status":   StatusStarted,
+			"meeting_topic":    "Webex Meeting",
+			"starting_user_id": userId,
+		},
+	}
+
+	createdPost, appErr := p.API.CreatePost(joinPost)
```

Comment:
Can you join the webex meeting before the host starts it?

Reply:
Yes, you'll just be waiting in the 'lobby' until it starts.